rocAL/rocAL_pybind/amd/rocal/plugin/tf.py

```diff
@@ -75,9 +78,17 @@ def __init__(self, pipeline, tensor_layout = types.NCHW, reverse_channels = Fals
             data_type="float16"
         
         if(types.NHWC == self.tensor_format):
-            self.out = np.zeros(( self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype = data_type)
-        else: 
-            self.out = np.zeros(( self.bs*self.n, self.p, int(self.h/self.bs), self.w), dtype = data_type)
+            if self.device == "cpu":
+                self.out = np.zeros(( self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype = data_type)
+            else:
+                with cp.cuda.Device(device=self.device_id):
+                    self.out = cp.zeros(( self.bs * self.n, int(self.h/self.bs), self.w, self.p), dtype = data_type)
+        else:
+            if self.device == "cpu":
+                self.out = np.zeros(( self.bs*self.n, self.p, int(self.h/self.bs), self.w), dtype = data_type)
+            else:
+                with cp.cuda.Device(device=self.device_id):
```

Comment:
Have we tested tf gpu backend with these changes ?
How are we converting the cupy arrays to tf tensors , is this feasible ?

Reply:
Yes, i have tested tf with GPU backend
I have made changes in the example file to handle cupy array